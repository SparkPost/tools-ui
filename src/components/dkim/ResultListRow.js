import React from 'react';
import { Link } from 'react-router';
import Icon from 'components/Icon';

import './ResultListRow.scss';

const ResultListRow = (props) => {
  const { id, subject, result, header_from, received, email } = props;

  return (
    <Link to={`/dkim/results/${email}/${id}`} className='dkimResultListRow'>
      <div className='panel marginBottom--none'>
        <div className='panel__body'>
          <div className={`dkimResultListRow__result ${result === 'Failed' && 'has-error'}`}>
            <Icon name={result === 'Passed' ? 'check-circle' : 'exclamation-circle'} />
          </div>
          <div className='dkimResultListRow__details'>
            <div className='dkimResultListRow__subject'>
              <span className='dkimResultListRow__label'>subject</span>
              {subject}
            </div>
            <div className='dkimResultListRow__from'>
              <span className='dkimResultListRow__label'>from</span>
              {header_from}
            </div>
            <div className='dkimResultListRow__received'>Delivered on {received} <Icon name='chevron-right' /></div>
          </div>
        </div>
      </div>
    </Link>
  );
};

ResultListRow.propTypes = {
  id: React.PropTypes.number.isRequired,
  subject: React.PropTypes.string.isRequired,
  result: React.PropTypes.string.isRequired,
  header_from: React.PropTypes.string.isRequired,
  received: React.PropTypes.string.isRequired
};

export default ResultListRow;
