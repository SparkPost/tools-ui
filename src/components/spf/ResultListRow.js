import React, { Component } from 'react';
import { Link } from 'react-router';
import Icon from 'components/Icon';

class ResultListRow extends Component {
  render() {
    const { domain, timestamp } = this.props;

    return (
      <div>
        <div>
          {this.renderStatus()}
        </div>
        <div>
          {domain}
          <span>Inspected on {timestamp}</span>
        </div>
        <div>
          <Link to={`/spf/results/${domain}`}>
            Refresh
          </Link>
        </div>
      </div>
  );
  }

  renderStatus() {
    const { status } = this.props;
    let iconCls = '';
    switch (status) {
      case 'valid':
        iconCls = 'check-status';
        break;
      case 'warning':
        iconCls = 'warning';
        break;
      case 'error':
        iconCls = 'exclamation-circle';
        break;
    }
    return (<Icon name={iconCls} />);
  }
}

ResultListRow.propTypes = {
  domain: React.PropTypes.string.isRequired,
  status: React.PropTypes.string.isRequired,
  timestamp: React.PropTypes.string.isRequired
};

export default ResultListRow;
