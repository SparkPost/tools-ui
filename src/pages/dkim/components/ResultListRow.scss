@import "~@sparkpost/styles/src/assets/scss/shared/config";
@import "~@sparkpost/styles/src/assets/scss/shared/functions";

.dkimResultListRow {
  position: relative;

  .panel { transition: box-shadow 0.15s; }
  .fa { transition: color 0.15s; }

  &:hover {
    text-decoration: none;

    .panel {
      position: relative;
      z-index: 1;
      box-shadow: shadow(2), 0 0 36px rgba(#000, 0.05);
    }

    .dkimResultListRow__action .fa {
      color: color(blue);
    }
  }

  &__result {
    position: absolute;
    top: 50%;
    left: rem(18);
    transform: translate(0, -50%);

    color: color(green);
  }

  &__result.has-error {
    color: color(alert);
  }

  &__details {
    padding-left: rem(30);
    line-height: rem(21);
    font-size: rem(14);
    color: grayscale(2);
    font-weight: 500;
  }

  &__label {
    display: inline-block;
    width: rem(42);
    margin-right: rem(18);

    text-align: right;
    font-size: rem(12);
    color: grayscale(5);
  }

  &__action {
    position: absolute;
    top: 50%;
    right: rem(18);
    transform: translate(0, -50%);

    font-size: rem(12);
    color: grayscale(5);
    font-weight: 500;

    .fa {
      display: inline-block;
      margin-left: rem(12);
      font-size: rem(15);
    }
  }

  &__received {
    display: none;
    @media screen and (min-width: break(md)) {
      display: inline;
    }
  }
}
