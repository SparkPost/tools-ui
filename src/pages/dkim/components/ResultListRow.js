import React from 'react';
import { Link } from 'react-router';
import Icon from 'components/Icon';

import './ResultListRow.scss';

const ResultListRow = (props) => {
  const { id, subject, result, header_from, received, email } = props;
  const resultClasses = result ? 'is-valid' : 'has-error';

  return (
    <Link to={`/dkim/results/${email}/${id}`} className='dkimResultListRow'>
      <div className='panel marginBottom--none'>
        <div className='panel__body'>
          <div className={`dkimResultListRow__result ${resultClasses}`}>
            <Icon name={result ? 'check-circle' : 'exclamation-circle'} />
          </div>
          <div className='dkimResultListRow__details'>
            <div className='dkimResultListRow__subject'>
              <span className='dkimResultListRow__label'>subject</span>
              {subject}
            </div>
            <div className='dkimResultListRow__from'>
              <span className='dkimResultListRow__label'>from</span>
              {header_from}
            </div>
            <div className='dkimResultListRow__action'>
              <span className='dkimResultListRow__received'>{received}</span>
              <Icon name='chevron-right' />
            </div>
          </div>
        </div>
      </div>
    </Link>
  );
};

ResultListRow.propTypes = {
  id: React.PropTypes.number.isRequired,
  subject: React.PropTypes.string.isRequired,
  result: React.PropTypes.bool.isRequired,
  header_from: React.PropTypes.string.isRequired,
  received: React.PropTypes.string.isRequired
};

export default ResultListRow;
