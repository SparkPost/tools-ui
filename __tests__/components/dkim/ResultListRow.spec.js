import React from 'react';
import renderer from 'react-test-renderer';

import ResultListRow from 'components/dkim/ResultListRow';

describe('ResultListRow component', () => {

  const testDate = new Date('2017-01-01T00:00Z').toString();

  test('should render', () => {
    expect(renderer.create(<ResultListRow id={5} subject='subject' result={true} header_from='some@email.com' received={testDate} email='some@email.com' />)).toMatchSnapshot();
  });

});
