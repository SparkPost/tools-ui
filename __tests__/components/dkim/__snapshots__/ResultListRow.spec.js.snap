exports[`ResultListRow component should render 1`] = `
<a
  className="dkimResultListRow"
  onClick={[Function]}
  style={Object {}}>
  <div
    className="panel marginBottom--none">
    <div
      className="panel__body">
      <div
        className="dkimResultListRow__result">
        <i
          className="fa fa-check-circle" />
      </div>
      <div
        className="dkimResultListRow__details">
        <div
          className="dkimResultListRow__subject">
          <span
            className="dkimResultListRow__label">
            subject
          </span>
          subject
        </div>
        <div
          className="dkimResultListRow__from">
          <span
            className="dkimResultListRow__label">
            from
          </span>
          some@email.com
        </div>
        <div
          className="dkimResultListRow__received">
          Delivered on 
          Sat Dec 31 2016 19:00:00 GMT-0500 (EST)
           
          <i
            className="fa fa-chevron-right" />
        </div>
      </div>
    </div>
  </div>
</a>
`;
